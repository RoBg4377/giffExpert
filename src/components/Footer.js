
export const Footer = () => {
  return (
    <div className="footer">
      <div className="iconos">
        <a
          href="https://github.com/settings/profile"
          target="noopener"
        >
          <i class="fa-brands fa-github-alt"></i>
        </a>
        <a
          href="https://linkedin.com/in/rodrigo-benavides-697136255"
          target='noopener noreferre'
        >
          <i class="fa-brands fa-linkedin"></i>
        </a>
        <a
          href="https://www.facebook.com/rbg4377" 
          target="noopener noreferrer"
        >
          <i class="fa-brands fa-facebook"></i>
        </a>
        <a
          href="https://www.instagram.com/rbg4377/"
          target="noopener noreferrer"
        >
          <i class="fa-brands fa-square-instagram"></i>
        </a>
      </div>
    </div>
  );
};
