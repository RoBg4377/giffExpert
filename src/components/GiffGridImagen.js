import React from 'react'

import PropTypes from "prop-types";

 export const GiffGridImagen = ({title, images}) => {

    return (
        <div className='card animate__animated animate__fadeInDown' >
        <img  src={images} alt={title} />
        <p>{title}</p>
    </div>    
  )
  
}
GiffGridImagen.propTypes = {
  title: PropTypes.string.isRequired,
  images: PropTypes.string.isRequired
}
