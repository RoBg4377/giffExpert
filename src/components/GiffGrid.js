import { GiffGridImagen } from "./GiffGridImagen";

import { useGiffsFech } from "../hooks/useGiffsFech";

export const GiffGrid = ({ cat }) => {
  const { data: gifs, loading } = useGiffsFech(cat);

  return (
    <>
      <h3>Imagenes Recibidas: {cat}</h3>

      {loading && (

        <div>
        <div class="spinner-grow cargador" role="status">
          <span class="visually-hidden">Loading...</span>
        </div>
        <div class="spinner-grow cargador" role="status">
          <span class="visually-hidden">Loading...</span>
        </div>
        <div class="spinner-grow cargador" role="status">
          <span class="visually-hidden">Loading...</span>
        </div>
        <div class="spinner-grow cargador" role="status">
          <span class="visually-hidden">Loading...</span>
        </div>
        </div>
      )
      }

      <div className="card-grid">
        {gifs.map((gif) => (
          <GiffGridImagen key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
