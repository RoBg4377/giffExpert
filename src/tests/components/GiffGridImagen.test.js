import { shallow } from "enzyme"
import { GiffGridImagen } from "../../components/GiffGridImagen"

describe('análisis del componentes GiffGridImage', () => {

    const images = 'http://localhost/images.jpg'
    const title = 'imagenes'
    const wrapper = shallow(<GiffGridImagen title={title} images={images} />);

    it('el componente debe renderizarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    })

    it('el párrafo debe contener un string correspondiete al "title"', () => {

        const p = wrapper.find('p')

        expect(p.text().trim()).toBe(title)

    })

    it('el elmento img debe tener como atributo un src = al url de la imbagen y un title = a title', ()=> {

        // const imgProps = wrapper.find('img').props();
        const imgSrc = wrapper.find('img').prop('src');
        const imgAlt = wrapper.find('img').prop('alt');

        expect(imgSrc).toBe(images);
        expect(imgAlt).toBe(title);

    })

    it('el div debe tener la clase card animate__animated animate__fadeInDown', () => {

        const div = wrapper.find('div')
        const classNameDelWrapper = div.prop('className');
        const classNameEsperada = 'card animate__animated'

        expect(classNameDelWrapper.includes(classNameEsperada)).toBe(true) 
    })

})

