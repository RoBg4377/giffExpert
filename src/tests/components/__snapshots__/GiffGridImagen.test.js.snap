// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an√°lisis del componentes GiffGridImage el componente debe renderizarse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="imagenes"
    src="http://localhost/images.jpg"
  />
  <p>
    imagenes
  </p>
</div>
`;
